mpr-research-data.py

```diff
@@ -1,51 +1,187 @@
-import json
-import sys
-
+# LIBRARIES TO IMPORT
+# --------------------------------------------------------------------------
 import pandas as pd
 import sqlalchemy as sql
+import os
+import sys
+from google.cloud import storage
+
+
+# SETUP CONFIG VARIABLES
+# --------------------------------------------------------------------------
+targetBucketName = os.getenv('GCLOUD_BUCKET', 'mpr-research-data-uploads')
+
+numberOfMonths = os.getenv('NUMBER_OF_MONTHS', 4)
+
+defaultQueryFolder = os.getenv('QUERY_FOLDER', 'queries')
+queryTemplateDict = {'course': os.getenv('COURSE_QEURY', 'courseQuery.sql'),
+                     'retrieve': os.getenv('RETRIEVE_QUERY', 'retrieveQuery.sql')}
+
+
+# FUNCTIONS
+# --------------------------------------------------------------------------
+
+def getDBCreds():
+
+    dbCredsDefaultDict = {'NAME': None,
+                          'USER': None,
+                          'PASSWORD': None,
+                          'HOST': None,
+                          'PORT': 3306}
+    dbCredsDict = {}
+
+    allKeyPartsFound = True
+    for credPart in dbCredsDefaultDict:
+        dbCredsDict[credPart] = os.getenv(
+            'DB_' + credPart, dbCredsDefaultDict[credPart])
+
+        if not dbCredsDict[credPart]:
+            print(
+                f'Did not find .env variable for M-Write Peer Review production DB key: {credPart}')
+            allKeyPartsFound = False
+
+    if not allKeyPartsFound:
+        sys.exit('Exiting.')
+
+    return dbCredsDict
+
+
+def queryRetriever(queryName, queryModifier=False,
+                   queryFolder=defaultQueryFolder):
+    with open(os.path.join(queryFolder, queryName)) as queryFile:
+        queryLines = ''.join(queryFile.readlines())
+
+    if queryModifier:
+        queryLines = queryLines.format(queryModifier)
+
+    return queryLines
+
+
+def makeDBConnection(dbParams):
+
+    try:
+        connectString = f'mysql+pymysql://{dbParams["USER"]}:{dbParams["PASSWORD"]}@{dbParams["HOST"]}:{dbParams["PORT"]}/{dbParams["NAME"]}'
+        engine = sql.create_engine(connectString)
+        print('DB connection established.')
+        return engine
+
+    except Exception as e:
+        print(f'Error Message: {e}')
+        print('Failed to establish DB connection.')
+        sys.exit('Exiting.')
+
+
+def makeGCPConnection(gcpJSON):
+
+    try:
+
+        # I am not sure why this is how GCloud needs to accept a JSON String.
+        # Will look into alternatives.
+        with open('gcpAccessKey.json', 'w') as jsonFile:
+            jsonFile.write(gcpJSON)
+        client = storage.Client.from_service_account_json('gcpAccessKey.json')
+        os.remove('gcpAccessKey.json')
```

Comment:
Try `from_service_account_info()`…


```suggestion
        client = storage.Client.from_service_account_info(gcpJSON)
```

From docs (hopefully correct ones) at: https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1beta1.services.model_service.ModelServiceClient

Reply:
I have implemented that. I also made the .env file consistent in how the two key values are taken.